packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1453,6 +1463,32 @@ abstract class ResidentRunner extends ResidentHandlers {
     _reportedDebuggers = true;
   }
 
+  @override
+  void printHelp({required bool details, bool reloadIsRestart = false}) {
```

Comment:
Can we instead add `reloadIsRestart` as a getter on `ResidentRunner` that defaults to returning `false` and override it in the `ResidentWebRunner`? Then we don't need to add an additional parameter here and won't need to override the implementation.

Another option could be to just access the `debuggingOptions` directly here and perform the check from above (probably with an additional condition to ensure we're actually running against a web target) directly in this function:

```
debuggingOptions.buildInfo.ddcModuleFormat != DdcModuleFormat.ddc ||
debuggingOptions.buildInfo.canaryFeatures != true,
```

Reply:
I didn't really like any option here when I thought about it, since `reloadIsRestart` is a weird getter to have since it only exists for legacy web behavior, and introducing web-specific logic into `ResidentRunner` also felt like a code smell. I think the getter is better than overriding the method and adding an extra parameter though, so I went with that.